scripts/bench/stt-bench.py

```
@@ -0,0 +1,1508 @@
+def safe_extract(tarball: Path, dest: Path) -> None:
+    """Extract without letting a member escape `dest`.
+
+    Python 3.12 has `filter="data"` for this; the box runs 3.10, so the check
+    is explicit rather than assumed.
+    """
+    dest = dest.resolve()
+    with tarfile.open(tarball) as tar:
+        for member in tar.getmembers():
+            target = (dest / member.name).resolve()
+            if not str(target).startswith(str(dest) + os.sep) and target != dest:
+                raise RuntimeError(f"{tarball.name}: member escapes the target directory: {member.name}")
```

Ruff (0.16.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)